lib/service/connect.go

```diff
@@ -104,19 +111,22 @@ func (process *TeleportProcess) reconnectToAuthService(role types.SystemRole) (*
 	}
 }
 
-func (process *TeleportProcess) compareVersions(resp *proto.PingResponse) error {
+func (process *TeleportProcess) authServerTooOld(resp *proto.PingResponse) (bool, error) {
 	serverVersion, err := semver.NewVersion(resp.ServerVersion)
 	if err != nil {
-		return trace.Wrap(err)
+		return false, trace.Wrap(err)
 	}
 	teleportVersion, err := semver.NewVersion(teleport.Version)
 	if err != nil {
-		return trace.Wrap(err)
+		return false, trace.Wrap(err)
 	}
-	if math.Abs(float64(serverVersion.Major-teleportVersion.Major)) > 1 {
-		process.log.Warnf("Only versions %d and %d are supported, auth server is %s.", teleportVersion.Major, teleportVersion.Major-1, serverVersion.Major)
+
+	if serverVersion.Major < teleportVersion.Major {
+		process.log.Errorf("Only versions %d and greater are supported, auth server is version %d. To connect anyway pass the '--skip-version-check' flag.", teleportVersion.Major, serverVersion.Major)
```

Comment:
Do we want to log this even if the user provided "--skip-version-check"? If so, it might make more sense to change the logged message depending on whether it was provided.